web/src/billing/event_status.ts

```diff
@@ -1,10 +1,31 @@
 import $ from "jquery";
+import {z} from "zod";
 
 import * as loading from "../loading";
 
 import * as helpers from "./helpers";
 
-function update_status_and_redirect(status_message, redirect_to) {
+const stripe_response_schema = z.object({
```

Comment:
Are these changes directly related to the TypeScript migration? If not, I think it would be better to make a separate commit to introduce these changes.

Reply:
Also, these changes aren't exactly refactoring changes as well they only were made to support typescript migration only.